gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/ClientLibraryReflectConfigComposer.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.gapic.composer;
+
+import com.google.api.generator.gapic.model.GapicContext;
+import com.google.api.generator.gapic.model.Message;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ClientLibraryReflectConfigComposer {
+  private ClientLibraryReflectConfigComposer() {} // Disallow instance creation
+
+  public static List<ReflectConfig> generateReflectConfig(GapicContext context) {
+    List<String> allConfigs = new ArrayList<>();
+    context.messages().forEach((id, msg) -> allConfigs.addAll(calculateReflectConfigList(id, msg)));
+    return allConfigs.stream()
+        .distinct()
+        .sorted()
+        .map(ReflectConfig::new)
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  /** List all classes in the message that should have a reflect-config entry */
+  @VisibleForTesting
+  static List<String> calculateReflectConfigList(String id, Message message) {
+    final String name = formatNestedClasses(id);
```

Comment:
Can we add some comment regarding why the format has to be this way? Same thing for the nested enums format below. Maybe a link to a GraalVM doc?

Reply:
Yes -- I'll add some comments and make the wording more precise than it is right now.

I've also just verified that this format is required. Without it, native-image provides 50ish warnings that each look like this:
```
Warning: Could not resolve com.google.protobuf.DescriptorProtos.FieldDescriptorProto for reflection configuration. Reason: java.lang.ClassNotFoundException: com.google.protobuf.DescriptorProtos.FieldDescriptorProto.
...
Warning: Could not resolve com.google.showcase.v1beta1.Sequence.Response for reflection configuration. Reason: java.lang.ClassNotFoundException: com.google.showcase.v1beta1.Sequence.Response.
```

Reply:
Comment added. Unable to find this requirement explicitly in the GraalVM docs, only via behavior.